### PR TITLE
Posts, Post Types: Default wp_get_post_parent_id() to global

### PR DESCRIPTION
**Pull request by audrasjb**
Trac ticket: https://core.trac.wordpress.org/ticket/48358

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by audrasjb** [comment]
The failing test looks like an unrelated issue.

**Comment by hellofromtonya** [comment]
@audrasjb It's a remote request to wp.org that has timed out. There's a ticket open to mock each of these HTTP remote requests https://core.trac.wordpress.org/ticket/54420. Adding this test to that ticket's list.

But you're right. This failing test is not associated with this PR.

File: src/wp-includes/post.php [hunk 1]
```diff
@@ -7643,11 +7643,11 @@ function _publish_post_hook( $post_id ) {
  *
  * @since 3.1.0
  *
- * @param int|WP_Post $post Post ID or post object.
+ * @param int|WP_Post|null $post Optional. Post ID or post object.
```
[the remark this answers is not in the document]
Comment (audrasjb) [reply src/wp-includes/post.php:7646]: LGTM 👍

**Review by hellofromtonya — APPROVED** [review]
Ready for `commit` 👍

**Comment by audrasjb** [comment]
Tests are passing now 😉

File: src/wp-includes/post.php [hunk 2]
```diff
@@ -7643,11 +7643,11 @@ function _publish_post_hook( $post_id ) {
  *
  * @since 3.1.0
  *
- * @param int|WP_Post $post Post ID or post object.
+ * @param int|WP_Post|null $post Optional. Post ID or post object. Default null.
```
Comment (TimothyBJacobs) [inline src/wp-includes/post.php:7646]: I think we may want to expand on what `null` means here with some text similar to the parameter description from `get_post`.

Comment (peterwilsoncc) [reply src/wp-includes/post.php:7646]: Agree with Timothy, aside from `get_post()` which goes in to full details, the most common text appears to be:

```
 * @param WP_Post|int|null $post   Optional. Post ID or post object. Defaults to global $post.
```

**Comment by peterwilsoncc** [comment]
https://core.trac.wordpress.org/changeset/52194

**Pull request closed without merging**